04week/algorithms.js

```diff
@@ -10,22 +10,77 @@ function getRandomInt(min, max) {
 
 let arr = [];
 
-for (let i = 0; i < 1000; i++) {
-  arr.push(getRandomInt(0, 1000));
+for (let i = 0; i < 10; i++) {
+  arr.push(getRandomInt(0, 10));
 }
 
 function bubbleSort(arr) {
-  // Your code here
+  for (let i = 0; i < arr.length; i++) {
+    for (let n = i+1; n < arr.length; n++) {
+      if (arr[i] > arr[n]) {
+        let temp = arr[i];
+        arr[i] = arr[n];
+        arr[n] = temp;
+      }
+    }
+  }
 }
 
+
+
 function mergeSort(arr) {
-  // Your code here
+  if (arr.length < 2) return arr;
+
+  let middle = Math.floor(arr.length / 2);
+  let left = arr.slice(0,middle);
+  let right = arr.slice(middle,arr.length);
+
+  return merge(mergeSort(left),mergeSort(right));
+}
+
+function merge(lefty, righty) {
+  const result = [];
+  while (lefty.length && righty.length) {
+    if (lefty[0] <= righty[0]) {
+      result.push(lefty.shift());
+    } else {
+      result.push(righty.shift());
+    }
+  }
+  while (lefty.length) {
+    result.push(lefty.shift());
+  }
+  while (righty.length) {
+    result.push(righty.shift());
+  }
+  return result;
 }
 
-function binarySearch(arr, item) {
-  // Your code here
+
+const sortedArray = [1,2,3,4,5,6,7,8,9,10];
+
+function binarySearch(sortedArray, item) {
+
+  // struggling with the logic on this
+  // keep getting Maximum call stack size exceeded
+  // recursion isn't dropping out correctly
+  // also having issue with item being in larger half of array
+  let end = sortedArray.length-1;
+  let middle = Math.floor(sortedArray.length / 2);
+  if (item === sortedArray[middle]) {
```

Comment:
this condition never hits, I would check and make sure that you're comparing the right things